Umbraco-Heartcore/API-Documentation/Backoffice/Grid-Editors/index.md

```diff
@@ -0,0 +1,147 @@
+:::
+
+## JSON Schema
+
+A [JSON Schema](https://json-schema.org/) is used to describe the how the value is stored and returned by the REST and GraphQL API.
+
+This allows for advanced configurations where for example an [UDI](https://our.umbraco.com/Documentation/Reference/Querying/UDI-identifiers/) is converted to a URL.
+
+The default Schema looks like this
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "string"
+}
+
+```
+
+Since it has `"type": "string"` the API's will return the stored value as a string.
+
+If the editor stores an `object` we can use `"type": "object"` instead, now the API's will return the value as a JSON object.
+
+By also adding properties, the API will be more clever of how the data is returned.
+
+For example an image editor that stores an `UDI` in the `url` field and the alt text in the `altText` field like this
+
+```json
+{
+  "url": "udi://media/45d27aa9fcb446e48ef4a07d754d9c9d",
+  "altText": "An example image"
+}
+```
+
+Could have a schema like
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "type": "object",
+  "properties": {
+    "altText": {
+      "type": "string"
+    },
+    "url": {
+      "type": "string",
+      "format": "uri-reference"
+    }
+  },
+  "required": [
+    "url"
+  ]
+}
+```
+
+Notice how there's a `"format":·"uri-reference"` to the `url` property in the schema. Combined with `"type": "string"` the API's knows this is a URI and will try to parse the value stored as an `UDI` and if successful, return the URL of the referenced item.
+
+Currently the following combinations are supported:
+
+**type**: `string`<br>
+**format**: `uri-reference`<br>
+**value**: Content or Media `UDI`<br>
+**returned**: A URL to the item if exists.<br>
+
+**type**: `string`<br>
+**format**: `rich-text`<br>
+**value**: Rich Text, for example from the TinyMCE editor<br>
+**returned**: The value where all `a` tags with a `locallink` href and `img` tags with a `data-udi` attribute is replaced with the correct links to the items.<br>
+
+## Limitations and best practices
+
+Currently there are some functionality and components that does not work well in the preview pane. This includes integrations with the backoffice like pickers and the Rich Text Editor. They do however still work when inserted on a page.
+
+To make your custom editors less likely to break with future updates, do not use any of the backoffice javascript directly. Always use the [headless backoffice bridge](https://github.com/umbraco/Umbraco.Headless.Backoffice.Bridge).
+
+If the library is missing any functionality raise an issue on the [Heartcore issue tracker](https://github.com/umbraco/Umbraco.Heartcore.Issues/issues).
+
+Try to avoid relying on backoffice CSS-classes. Instead we recommend creating isolated elements using [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM).
```

Comment:
```suggestion
Try to avoid relying on backoffice CSS-classes. Instead, we recommend creating isolated elements using [shadow DOM](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM).
```